src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzServiceAccountConsumerStep.java

```diff
@@ -35,12 +41,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
 
     // Allow the steward and reader to bill this project to read requester pays bucket data.
     // The underlying service provides retries so we do not need to retry this operation
+    // Also grant the dataset's service account the permission to also access source buckets
```

Comment:
We are not granting this extra permission if we're using the general SA, has it been granted at some other point or does it not need it?